packages/bolt-connection/src/bolt/bolt-protocol-v3.js

```diff
@@ -78,7 +78,7 @@ export default class BoltProtocol extends BoltProtocolV2 {
     // passing notification filter on this protocol version throws an error
     assertNotificationFilterIsEmpty(notificationFilter, this._onProtocolError, observer)
 
-    this.write(RequestMessage.hello(userAgent, authToken), observer, true)
+    this.write(RequestMessage.hello(userAgent === '' || userAgent == null ? boltAgent : userAgent, authToken), observer, true)
```

Comment:
Should we treat `''` as null?

Reply:
if i dont it causes a testkit test to fail